builder/remotecontext/detect.go

```diff
@@ -30,6 +30,12 @@ func Detect(config backend.BuildConfig) (remote builder.Source, dockerfile *pars
 	switch {
 	case remoteURL == "":
 		remote, dockerfile, err = newArchiveRemote(config.Source, dockerfilePath)
+	case remoteURL == "client-session":
+		res, err := parser.Parse(config.Source)
```

Comment:
This matches the behavior if `remoteUrl` is set to a http url pointing directly to a dockerfile and could be used for other remote sources as well. For this specific case we could switch to use `SyncFrom` for getting the `Dockerfile` as well. That would give us the freedom to implement `INCLUDE` like behavior in the future. Currently was planning that as a follow-up.